pkgs/checks/lib/src/checks.dart

```diff
@@ -59,48 +59,57 @@ Check<T> checkThat<T>(T value, {String? because}) => Check._(_TestContext._root(
     value: _Present(value),
     // TODO - switch between "a" and "an"
     label: 'a $T',
-    reason: because,
-    fail: (m, _) => throw TestFailure(m),
+    fail: (f) {
+      final which = f.rejection.which;
+      throw TestFailure([
+        ...prefixFirst('Expected: ', f.detail.expected),
+        ...prefixFirst('Actual: ', f.detail.actual),
+        ...indent(['Actual: ${f.rejection.actual}'], f.detail.depth),
```

Comment:
it'd be nice to see a test showing the output here!

Reply:
I'll add a couple cases that show the full output of simple examples. https://github.com/dart-lang/test/pull/1820